### ISSUE TITLE
Focusing issues

### ISSUE THREAD

**Issue by olmokramer** [olmokramer/atom-autohide-tree-view#54]
A continuation of #28, where @lukescott and I have been discussing regarding focusing and de-focusing/blurring of the tree view. I'll try to collect the relevant parts of that thread here.

https://github.com/olmokramer/atom-autohide-tree-view/issues/28#issuecomment-141752520:

> The above issue I mentioned is fixed - If I open the dialog with Ctrl-\, right click and choose Search in Directory the focus goes to the search box and the tree-view stays open. So far so good. However, if I press Ctrl-\ again to dismiss while in the search box, the focus is restored to my editor. -- I would keep it exactly like it is with the tree-view remaining open until Ctrl-\ is pressed, but I would only restore the focus if the tree-view has it.

https://github.com/olmokramer/atom-autohide-tree-view/issues/28#issuecomment-141812146:

> I did a bit of debugging. I'm not sure if it's the context menu, but a mouse leave event does get triggered after choosing an item from the context menu. And a mouse enter is probably caused after. That would cause the focus issue with the rename dialog.

https://github.com/olmokramer/atom-autohide-tree-view/issues/28#issuecomment-142972558:

> The problem is that mouse even is causing in/out to make the tree-view refocus itself. This seems to be related to the context menu. There is a context menu event, so maybe we could use that somehow?

https://github.com/olmokramer/atom-autohide-tree-view/issues/28#issuecomment-143076473:

> How about we ignore the next mouseenter event in the contextmenu event, and then re-enable it again the next time the tree view hides. I'm not entirely sure this will work all the time, I'll think about it.

https://github.com/olmokramer/atom-autohide-tree-view/issues/28#issuecomment-143243652:

> Ok so it works as it did in the last commit - so rename works. Although when I use Ctrl-\ and try to search it also loses focus just like hover - unless I switch show on to something other than "hover", such as "none". So I still have a work-around for that at least.

@lukescott sounds about right?

**Comment by lukescott** [comment]
Yep that seems to be all the details.

**Comment by ProLoser** [comment]
I think I have to discontinue using this plugin because it's super buggy. I have to move the mouse to the edge of the screen multiple times or in just the right way to get it to open

**Comment by olmokramer** [comment]
@ProLoser Yeah I noticed that. It could happen when a show triggered too quickly after a hide. I've fixed that (done a rewrite of lots of stuff), but I want to try the plugin one more day to see if everything's working. You can check out the master branch if you want to try it yourself :)
